domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+// The subset of changes returned is maintained in the same order as they are received.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		modelUUIDs := make([]coremodel.UUID, len(changes))
+		for i, change := range changes {
+			modelUUIDs[i] = coremodel.UUID(change.Changed())
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There will be no errors returned if there are no activated models found.
+		if err != nil || len(activatedModelUUIDs) == 0 {
```

Comment:
When reading this it took me a few goes to appreciate how this worked. To me this suggest that it is complicated for no reason.

Can I suggest breaking these out into two if statements to help readability.

Reply:
Sure